src/types.ts

```diff
@@ -0,0 +1,25 @@
+export type ScopeStr = 'device' | 'local' | 'private' | 'public';
+
+export type Incoming = {
+  scope: ScopeStr | Array<ScopeStr>;
+  transform: 'shs' | 'noauth';
+  port?: number;
+  host?: string;
+};
```

Comment:
This specifies an object shape that:

- requires a `scope` (either a "scope string" or an array of scope strings)
- requires a `transform` (which is either the string `shs` or the string `noauth`, nothing else)
- can have a `port` (which is a number)
- can have a `host` (which is a string)